src/main/java/data/implementations/EdgeEntity.java

```diff
@@ -0,0 +1,84 @@
+package data.implementations;
+
+import data.IdGenerator;
+import data.model.Node;
+import data.model.Edge;
+
+public class EdgeEntity implements Edge {
+
+    private int id;
+    private int from;
+    private int to;
+    private int length;
+
+    public EdgeEntity(Node fromNode, Node toNode){
```

Comment:
Jeżeli mamy kilka konstruktorów, które działają podobnie i różnią się jedynie ilością parametrów, to można je wywołać kaskadowo.
```suggestion
    public EdgeEntity(Node fromNode, Node toNode){
        this(fromNode.getId(), toNode.getId(), calculateLength(fromNode,toNode))
    }
```
To powinno wywołać kolejny konstruktor, który przyjmuje tylko idFrom, idTo, oraz length